### DISCUSSION TITLE
How to setup second admin panel under the same domain but another path?

### DISCUSSION THREAD

**Discussion by mialex** [Laravel-Backpack/community-forum#1048] [Q&A (Help)] [answered]
I have an admin, which is working fine. Now I want to have another simple admin (portal) for regular users to see other details. For example in e-commerce.
1. There is an Admin for administrators who create categories, and products and see all orders. it's done with Backpack
2. There is a web page with all production where users make their purchases. Not Backpack, but something custom

I want one more Portal for users which should be done with Backpack, where users, for ex. can see their history, orders, etc.
The idea is to share the same project, but just have other routes, controllers, and maybe view, but use the same services, models, etc...

I don't want to move shared code to the library and create 2 separate projects Admin and Portal, I would like to have then under the same project. Is this possible?

**Comment by karandatwani92** [comment, marked as answer]
Hey @mialex 

Let's say your current routes setup is like:
- .env: `APP_URL = domain.test`
- config/backpack/base.php: `route_prefix` = `admin`
- Controllers's namespace: `App\Http\Controllers\Admin;`
- `https://domain.test/admin/order`

And In the demo, you can notice a **pet-shop**:
- https://demo.test/admin/pet-shop/invoice
- It's Route setup is [here](https://github.com/Laravel-Backpack/demo/blob/main/routes/backpack/custom.php#L76) and its [controllers](https://github.com/Laravel-Backpack/demo/blob/ea5765fa4b3ef651f7fb6fb2ffbbc94e7d0f6987/app/Http/Controllers/Admin/PetShop/InvoiceCrudController.php#L3) are put in sub-namespace `App\Http\Controllers\Admin\PetShop`


In the same way, I recommend you create separate routes & CRUD controllers for each role. To change the word `admin` from the URL, you change`route_prefix`. For example:
- You change `route_prefix`  = ~~admin~~ `shop`.
- Create CRUDs for users in `App\Http\Controllers\Backpack\User` namespace.
- Move CRUDs of admins to `App\Http\Controllers\Backpack\Admin` namespace.

After, your final URLs would be:
- `https://domain.test/shop/user/order`
- `https://domain.test/shop/admin/order`
- `https://{domain}/{route_prefix}/{namespace-or-role-resembling-keyword}/{entity}`

For Implementing User Access Control and Permissions, check this [article](https://backpackforlaravel.com/articles/tutorials/guide-on-access-control-for-your-admin-panel).

**Reply by mialex** [reply]
Thank you for such a detailed explanation. It was also my thought to implement it the same way. The only thing that I didn't want to have was this common route prefix.

In my case, I decided to create a mono repo and move shared components like `Service`, `Model`, `Provider`, etc to the library and create 2 separate projects with their own configs and `route_prefix`. In my case, I see here more benefits :)

The only 2 issues I need to solve are:
1. How to move custom views to the library and use them in my projects, as I have created or redefined some views
2. My Save action button from the dropdown stopped working. I can click the button "Save and back", but when I choose from the drop-down like "Save and edit", it doesn't work :)

**Reply by karandatwani92** [reply]
1. Breaking the project into smaller packages is a nice idea. Here is a [guide](https://backpackforlaravel.com/docs/6.x/add-ons-tutorial-using-the-addon-skeleton#part-b-create-the-package) to help. Info on view files moving is [here](https://backpackforlaravel.com/docs/6.x/add-ons-tutorial-using-the-addon-skeleton#files-inside-your-projects-resources-views-directory).

2. No idea! What it could. Please try the above, and we will talk about it later in another discussion.
